src/util/ByteOrder.hxx

```diff
@@ -516,7 +517,61 @@ public:
 			x = ByteSwap64(x);
 		return x;
 	}
-};
+} __attribute__((packed));
 
 static_assert(sizeof(PackedLE64) == sizeof(uint64_t), "Wrong size");
 static_assert(alignof(PackedLE64) == 1, "Wrong alignment");
+
+/**
+ * Classes for floats rely on assumption that the platform uses IEEE754
+ * floating point representation (aso known as IEC559)
+ */
+static_assert(std::numeric_limits<float>::is_iec559);
+
+/**
+ * Float represented by IEEE754 standard. With bytes swapped (little-endian)
+ */
+class PackedFloatLE
+{
+  uint8_t a, b, c, d;
+
+public:
+  PackedFloatLE() = default;
+
+  constexpr PackedFloatLE(float f)
+  {
+    uint32_t value_as_uint;
+    value_as_uint = *reinterpret_cast<uint32_t*>(&f);
```

Comment:
This is the worst possible way to express this because the compiler is forced to allocate stack space for the parameter (cannot be an a register) and it violates C++'s aliasing rules, so it's UB.